internal/services/pubkey_service.go

```diff
@@ -99,6 +109,50 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkey, err := pubkeyDao.GetById(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err))
```

Comment:
probably a leftover? :)
```suggestion
		renderError(w, r, payloads.NewNotFoundError(r.Context(), err))
```

or ideally 
```suggestion
		if errors.Is(err, dao.NoRowsError) {
		  renderError(w, r, payloads.NewNotFoundError(r.Context(), err))
		} else {
		  renderError(w, r, payloads.NewDAOError(r.Context(), "pubkey DAO", err))
		}
```